datadog_lambda/wrapper.py

```diff
@@ -105,24 +118,59 @@ def __call__(self, event, context, **kwargs):
 
     def _before(self, event, context):
         try:
+
             set_cold_start()
             submit_invocations_metric(context)
             # Extract Datadog trace context from incoming requests
-            extract_dd_trace_context(event)
+            dd_context = extract_dd_trace_context(event)
+
+            self.span = None
+            if dd_tracing_enabled:
+                self.span = self._create_dd_trace_py_span(context, dd_context)
+            else:
+                set_correlation_ids()
 
-            # Set log correlation ids using extracted trace context
-            set_correlation_ids()
             logger.debug("datadog_lambda_wrapper _before() done")
         except Exception:
             traceback.print_exc()
 
     def _after(self, event, context):
         try:
+            if self.span:
+                self.span.finish()
             if not self.flush_to_log:
                 lambda_stats.flush(float("inf"))
             logger.debug("datadog_lambda_wrapper _after() done")
         except Exception:
             traceback.print_exc()
 
+    def _create_dd_trace_py_span(self, context, trace_context):
```

Comment:
Nits, can we move this function to tracing.py and pass in needed `self.*` attributes as parameters, so tracing related code all live in the same place?

Reply:
Or define this logic as a class in tracing.py and mix it into the wrapper class here?

Reply:
I've moved this logic out into the tracing file